core/src/main/kotlin/org/jetbrains/academy/test/system/core/MethodUtils.kt

```diff
@@ -39,9 +39,9 @@ fun Array<Method>.findMethod(method: TestMethod, customErrorMessage: String? = n
                 it.name == method.name
             }
         }
-    val returnTypeJava = (method.returnTypeJava?.let { listOf(it) } ?: listOfNotNull(method.returnType.type, *method.returnType.possibleBounds.toTypedArray())).map { it.lowercase() }
+    val returnTypeJava = (listOf(method.returnTypeJava)).map { it.lowercase() }
```

Comment:
But now you ignores the kotlin type at all...

Reply:
I think it can raises some errors in tests, the easiest way to check is to publish this version locally and check on this repo https://github.com/jetbrains-academy/kotlin-onboarding-introduction
Tests should pass

Reply:
Yes, thanks! I've reverted it back to the previous state. 
To work for java, `TestKotlinType` is created in the constructor.